doc/api.md

```diff
@@ -0,0 +1,81 @@
+# Microcluster REST API
+
+Microcluster offers a default REST API called the Core API for managing a cluster.
+
+The Core API is offered over the local unix socket as well as over the cluster address of initialized clusters.
+
+There are 4 components to the Core API:
+
+* `/core/internal`: The prefix for all internal endpoints used internally by Microcluster.
+* `/core/control`: The prefix for all internal endpoints available exclusively over the unix socket.
+* `/core/1.0`: The prefix for all stable endpoints that are available for direct external use.
+* `/*`: Extensions to the Core API provided by the external project.
+
+Any additions to the internal Microcluster API will happen under the `/core` endpoint path prefix. This path prefix is reserved and should not be used for additionally supplied API resources.
+
+
+## Additional servers
+
+Microcluster can set up additional servers with `DaemonArgs.Servers`:
+
+```go
+dargs := microcluster.DaemonArgs{
+    // If specified, a listener will start, using the `server` keypair, and offer all extensions to the CoreAPI as well as `/core/1.0`.
+    // Once the daemon is bootstrapped or joins a cluster, this listener will turn off permanently.
+    PreInitListenAddress: "10.0.0.100:8000",
+
+    // Set of servers to set up with the Core API.
+    Servers: map[string]Server{
+        // Unique internal name for the server.
+        "extend-core-api": {
+            // Don't set up an additional server, instead serve all resources over the Core API.
+            CoreAPI: true,
+
+            // Allow resources to be available over the unix socket or PreInitListenAddress before bootstrapping or joining a cluster.
+            PreInit: true,
+
+            // Allow resources to be served over the local unix socket.
+            ServeUnix: true,
+
+            // Default address to serve the additional listener over.
+            //
+            // Cannot be used with CoreAPI=true, because the CoreAPI address and port will be used.
+            ServerConfig: types.ServerConfig{},
```

Comment:
Could this be moved to `additional-api` below, since it can't be used with `CoreAPI: true`?

Reply:
I deliberately initialized every key in the first example so I could explain why they're not used for the CoreAPI server.